### ISSUE TITLE
b.pageCount() as setter

### ISSUE THREAD

**Issue by trych** [basiljs/basil.js#136]
Small improvement: `b.pageCount()` could also act as setter (it's a getter only at the moment). Most getters work as setters in basil, would be nice, if the same worked for `b.pageCount()`.

```js
b.pageCount(20); // -> changes document to have 20 pages
```

**Comment by fabianmoronzirfas** [comment]
What happens if I set a number lower then my current page count? Will it delete pages?

**Comment by fabianmoronzirfas** [comment]
How does it handle master spreads?

**Comment by trych** [comment]
I think it should just adopt the behavior of `b.doc().documentPreferences.pagesPerDocument = n`.

> What happens if I set a number lower then my current page count? Will it delete pages?

Yes, if I add a line `b.pageCount(5)` at a point in a script where I know I have a 20 page document, I should expect to have pages 6 - 20 deleted.

> How does it handle master spreads?

Just as `b.doc().documentPreferences.pagesPerDocument = n` would as well. I believe the behavior is to have the master page of the currently active page applied to the new pages.

I just think it is a simple addition that would be mostly used in document setup. As beginners don't know about (the very non-basil) `b.doc().documentPreferences.pagesPerDocument` the standard way of getting a 20 page document seems to be:

```js
for(var i = 1; i < 20; i++) {
    b.addPage();
}
```

So as `b.pageCount()` already exists anyway, it would be nice to be able to just do `b.pageCount(20)` instead.

**Comment by ffd8** [comment]
Seems a tad dangerous that it can just remove pages by giving the potentially wrong number (counting accidents happen.. especially with beginners when 0 usually equals the new 1). 

I wonder if such functionality should be merged to the `b.addPage()` and `b.removePage()`? Unfortunately they were built with the single page in mind.. rather than `b.addPages(#, where)`. There are a few functions that already depend on the singular vs plural wording `b.selection()` vs `b.selections()`.. so that could be an option?

**Comment by trych** [comment]
> Seems a tad dangerous that it can just remove pages by giving the potentially wrong number

Hm, I cannot really imagine an actually dangerous real life scenario here, it's not like people run this command on their production files for testing and then save those right afterwards.

If however during script development they figure out that more pages are deleted than intended, then they can simply correct this error until they got it right and then when the script is behaving as intended use it for "production".

I'm pretty sure 99% of the time the feature would be used during "document setup stage" to just give your initial blank document the number of pages you want it to have.

Again: I think the behavior to use this not only as a getter but also as a setter would make it more consistent with other basil functions.

**Comment by b-g** [comment]
Hi @trych, cc @ffd8 

I would vote to implement the `b.pageCount(#);` as @trych suggest (also with removing/adding pages).

We had similar discussion wether we should really give the users a full/unlimited `b.shellExecute()` = what could be more dangerous than this!? As even with this high potential harm function no bad things happened ... touch wood ... I'm not at all concerned!

**Comment by trych** [comment]
This has been included via #209.
